File: app/src/main/java/com/giosis/util/qdrive/settings/NoticeDetailActivity.java
package com.giosis.util.qdrive.settings;

import android.app.ProgressDialog;
import android.content.Context;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.giosis.util.qdrive.singapore.MyApplication;
import com.giosis.util.qdrive.singapore.R;
import com.giosis.util.qdrive.util.Custom_JsonParser;
import com.giosis.util.qdrive.util.DataUtil;
import com.giosis.util.qdrive.util.DisplayUtil;
import com.giosis.util.qdrive.util.NetworkUtil;
import com.giosis.util.qdrive.util.SharedPreferencesHelper;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;

/**
 * @author krm0219
 **/
public class NoticeDetailActivity extends AppCompatActivity {
    String TAG = "NoticeDetailActivity";


    FrameLayout layout_top_back;
    TextView text_top_title;

    LinearLayout layout_notice_detail;
    TextView text_notice_detail_content;
    TextView text_notice_detail_date;
    LinearLayout layout_notice_detail_prev;
    LinearLayout layout_notice_detail_next;

    LinearLayout layout_notice_detail_reload;
    Button btn_notice_detail_reload;

    Context context;
    String opID;
    String officeCode;

    String noticeNo;
    String prevNoticeNo = null;
    String nextNoticeNo = null;

    ProgressDialog progressDialog = null;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_notice_detail);

        layout_top_back = findViewById(R.id.layout_top_back);
        text_top_title = findViewById(R.id.text_top_title);

        layout_notice_detail = findViewById(R.id.layout_notice_detail);
        text_notice_detail_content = findViewById(R.id.text_notice_detail_content);
        text_notice_detail_date = findViewById(R.id.text_notice_detail_date);
        layout_notice_detail_prev = findViewById(R.id.layout_notice_detail_prev);
        layout_notice_detail_next = findViewById(R.id.layout_notice_detail_next);

        layout_notice_detail_reload = findViewById(R.id.layout_notice_detail_reload);
        btn_notice_detail_reload = findViewById(R.id.btn_notice_detail_reload);


        //
        text_top_title.setText(R.string.text_title_notice);

        layout_top_back.setOnClickListener(clickListener);
        layout_notice_detail_prev.setOnClickListener(clickListener);
        layout_notice_detail_next.setOnClickListener(clickListener);
        btn_notice_detail_reload.setOnClickListener(clickListener);

        progressDialog = new ProgressDialog(NoticeDetailActivity.this);
        context = getApplicationContext();
//        opID = SharedPreferencesHelper.getSigninOpID(getApplicationContext());
//        officeCode = SharedPreferencesHelper.getSigninOfficeCode(getApplicationContext());
        opID = MyApplication.preferences.getUserId();
        officeCode = MyApplication.preferences.getOfficeCode();

        noticeNo = getIntent().getStringExtra("notice_no");
        Log.e("krm0219", TAG + "  notice No > " + noticeNo);

        NoticeDetailAsyncTask noticeDetailAsyncTask = new NoticeDetailAsyncTask(noticeNo);
        noticeDetailAsyncTask.execute();
    }


    public class NoticeDetailAsyncTask extends AsyncTask<Void, Void, NoticeResult> {

        String noticeNo;

        NoticeDetailAsyncTask(String noticeNo) {

            this.noticeNo = noticeNo;
        }

        @Override
        protected void onPreExecute() {
            super.onPreExecute();

            progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
            progressDialog.setMessage(context.getResources().getString(R.string.text_please_wait));
            progressDialog.setCancelable(false);
            progressDialog.show();
        }

        @Override
        protected NoticeResult doInBackground(Void... params) {

            NoticeResult result = new NoticeResult();

            if (!NetworkUtil.isNetworkAvailable(context)) {

                result.setResultCode("-16");
                result.setResultMsg(context.getResources().getString(R.string.msg_network_connect_error_saved));

                return result;
            }

            try {

                JSONObject job = new JSONObject();
                job.accumulate("gubun", "DETAIL");
                job.accumulate("kind", "QSIGN");
                job.accumulate("page_no", 0);
                job.accumulate("page_size", 0);
                job.accumulate("nid", noticeNo);
                job.accumulate("svc_nation_cd", "SG");
                job.accumulate("opId", opID);
                job.accumulate("officeCd", officeCode);
                job.accumulate("app_id", DataUtil.appID);
                job.accumulate("nation_cd", DataUtil.nationCode);

                String methodName = "GetNoticeData";
                String jsonString = Custom_JsonParser.requestServerDataReturnJSON(methodName, job);
                // {"ResultObject":[{"total_cnt":null,"nid":"158577","kind":"","title":"hello","link":"","priority":"","reg_dt_short":"Apr 3","reg_dt_long":"Apr 3, 2:02 PM","rownum":null,"contents":"","nextnid":"158578","prevnid":"158575"}],"ResultCode":0,"ResultMsg":"SUCCESS"}

                JSONObject jsonObject = new JSONObject(jsonString);
                JSONArray resultArray = jsonObject.getJSONArray("ResultObject");
                JSONObject resultObject = resultArray.getJSONObject(0);

                String resultCode = jsonObject.getString("ResultCode");
                String resultMsg = jsonObject.getString("ResultMsg");

                result.setResultCode(resultCode);
                result.setResultMsg(resultMsg);

                ArrayList<NoticeResult.NoticeListItem> listItemArrayList = new ArrayList<>();

                NoticeResult.NoticeListItem noticeListItem = new NoticeResult.NoticeListItem();
                noticeListItem.setNoticeNo(resultObject.getString("nid"));
                noticeListItem.setNoticeDate(resultObject.getString("reg_dt_long"));
                noticeListItem.setNoticeContent(resultObject.getString("title"));
                noticeListItem.setPrevNo(resultObject.getString("prevnid"));
                noticeListItem.setNextNo(resultObject.getString("nextnid"));
                listItemArrayList.add(noticeListItem);

                result.setResultObject(listItemArrayList);
            } catch (Exception e) {

                Log.e("Exception", TAG + "  GetNoticeData Exception : " + e.toString());

                String msg = String.format(context.getResources().getString(R.string.text_exception), e.toString());
                result.setResultCode("-15");
                result.setResultMsg(msg);
            }

            return result;
        }

        @Override
        protected void onPostExecute(NoticeResult result) {
            super.onPostExecute(result);

            DisplayUtil.dismissProgressDialog(progressDialog);

            if (result.getResultCode().equals("0")) {

                layout_notice_detail.setVisibility(View.VISIBLE);
                layout_notice_detail_reload.setVisibility(View.GONE);

                text_notice_detail_content.setText(result.getResultObject().get(0).getNoticeContent());
                text_notice_detail_date.setText(result.getResultObject().get(0).getNoticeDate());


                if (result.getResultObject().get(0).getNextNo().length() == 0) {

                    layout_notice_detail_next.setVisibility(View.GONE);
                } else {

                    layout_notice_detail_next.setVisibility(View.VISIBLE);
                    nextNoticeNo = result.getResultObject().get(0).getNextNo();
                }

                if (result.getResultObject().get(0).getPrevNo().length() == 0) {

                    layout_notice_detail_prev.setVisibility(View.GONE);
                } else {

                    layout_notice_detail_prev.setVisibility(View.VISIBLE);
                    prevNoticeNo = result.getResultObject().get(0).getPrevNo();
                }
            } else {

                layout_notice_detail.setVisibility(View.GONE);
                layout_notice_detail_reload.setVisibility(View.VISIBLE);
            }
        }
    }


    View.OnClickListener clickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {

            switch (v.getId()) {
                case R.id.layout_top_back: {

                    finish();
                }
                break;

                case R.id.layout_notice_detail_prev: {

                    noticeNo = prevNoticeNo;

                    NoticeDetailAsyncTask noticeDetailAsyncTask = new NoticeDetailAsyncTask(noticeNo);
                    noticeDetailAsyncTask.execute();
                }
                break;

                case R.id.layout_notice_detail_next: {

                    noticeNo = nextNoticeNo;

                    NoticeDetailAsyncTask noticeDetailAsyncTask = new NoticeDetailAsyncTask(noticeNo);
                    noticeDetailAsyncTask.execute();
                }
                break;

                case R.id.btn_notice_detail_reload: {

                    NoticeDetailAsyncTask noticeDetailAsyncTask = new NoticeDetailAsyncTask(noticeNo);
                    noticeDetailAsyncTask.execute();
                }
                break;
            }
        }
    };
}

File: appMY/src/main/java/com/giosis/util/qdrive/settings/NoticeDetailActivity.kt
package com.giosis.util.qdrive.settings

import android.annotation.SuppressLint
import android.os.AsyncTask
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.ProgressBar
import com.giosis.util.qdrive.barcodescanner.ManualHelper
import com.giosis.util.qdrive.international.MyApplication
import com.giosis.util.qdrive.international.R
import com.giosis.util.qdrive.util.Custom_JsonParser
import com.giosis.util.qdrive.util.DataUtil
import com.giosis.util.qdrive.util.NetworkUtil
import com.giosis.util.qdrive.util.ui.CommonActivity
import kotlinx.android.synthetic.main.activity_notice_detail.*
import kotlinx.android.synthetic.main.top_title.*
import org.json.JSONObject

class NoticeDetailActivity : CommonActivity() {

    val tag = "NoticeDetailActivity"
    private val context = MyApplication.getContext()
    private val userId = MyApplication.preferences.userId
    private val officeCode = MyApplication.preferences.officeCode

    private val progressBar = ProgressBar(context)

    lateinit var seqNo: String
    lateinit var seqNextNo: String
    lateinit var seqPrevNo: String


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_notice_detail)


        text_top_title.text = context.resources.getString(R.string.text_title_notice)

        progressBar.layoutParams = LinearLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        layout_notice_progress.addView(progressBar)
        layout_notice_progress.visibility = View.VISIBLE


        layout_top_back.setOnClickListener {

            finish()
        }

        layout_notice_detail_prev.setOnClickListener {

            seqNo = seqPrevNo
            val noticeDetailAsyncTask = NoticeDetailAsyncTask(seqNo)
            noticeDetailAsyncTask.execute()
        }

        layout_notice_detail_next.setOnClickListener {

            seqNo = seqNextNo
            val noticeDetailAsyncTask = NoticeDetailAsyncTask(seqNo)
            noticeDetailAsyncTask.execute()
        }

        btn_notice_detail_reload.setOnClickListener {

            val noticeDetailAsyncTask = NoticeDetailAsyncTask(seqNo)
            noticeDetailAsyncTask.execute()
        }


        seqNo = intent.getStringExtra("notice_no")
        val noticeDetailAsyncTask = NoticeDetailAsyncTask(seqNo)
        noticeDetailAsyncTask.execute()
    }


    @SuppressLint("StaticFieldLeak")
    inner class NoticeDetailAsyncTask(private val seqNo: String) : AsyncTask<Void, Void, NoticeResults>() {

        override fun doInBackground(vararg p0: Void?): NoticeResults {

            val result = NoticeResults()

            if (!NetworkUtil.isNetworkAvailable(context)) {

                result.resultCode = "-16"
                result.resultMsg = context.resources.getString(R.string.msg_network_connect_error_saved)

                return result
            }

            try {

                val job = JSONObject()
                job.accumulate("opId", userId)
                job.accumulate("officeCd", officeCode)
                job.accumulate("gubun", "DETAIL")
                job.accumulate("kind", "QSIGN")
                job.accumulate("page_no", 0)
                job.accumulate("page_size", 0)
                job.accumulate("nid", seqNo)
                job.accumulate("svc_nation_cd", DataUtil.nationCode)
                job.accumulate("app_id", DataUtil.appID)
                job.accumulate("nation_cd", DataUtil.nationCode)

                Log.e(tag, "$userId  /  $officeCode / $seqNo")

                val methodName = "GetNoticeData"
                val jsonString = Custom_JsonParser.requestServerDataReturnJSON(ManualHelper.MOBILE_SERVER_URL, methodName, job)
                // {"ResultObject":[{"total_cnt":null,"nid":"158577","kind":"","title":"hello","link":"","priority":"","reg_dt_short":"Apr 3","reg_dt_long":"Apr 3, 2:02 PM","rownum":null,"contents":"","nextnid":"158578","prevnid":"158575"}],"ResultCode":0,"ResultMsg":"SUCCESS"}

                val jsonObject = JSONObject(jsonString)
                result.resultCode = jsonObject.getString("ResultCode")
                result.resultMsg = jsonObject.getString("ResultMsg")

                val jsonArray = jsonObject.getJSONArray("ResultObject")
                val resultObject = jsonArray.getJSONObject(0)

                val noticeList = arrayListOf<NoticeResults.NoticeItem>()
                val noticeItem = NoticeResults.NoticeItem()

                noticeItem.seqNo = resultObject.getString("nid")
                noticeItem.date = resultObject.getString("reg_dt_long")
                noticeItem.content = resultObject.getString("title")
                noticeItem.prevNo = resultObject.getString("prevnid")
                noticeItem.nextNo = resultObject.getString("nextnid")
                noticeList.add(noticeItem)

                result.resultObject = noticeList
            } catch (e: Exception) {

                Log.e("krm0219", "Exception : $e")
                result.resultCode = "-15"
                result.resultMsg = java.lang.String.format(context.resources.getString(R.string.text_exception), e.toString())
            }

            return result
        }


        override fun onPostExecute(result: NoticeResults) {
            super.onPostExecute(result)

            progressBar.visibility = View.GONE

            if (result.resultCode == "0") {

                layout_notice_detail.visibility = View.VISIBLE
                layout_notice_detail_reload.visibility = View.GONE

                text_notice_detail_content.text = result.resultObject[0].content
                text_notice_detail_date.text = result.resultObject[0].date


                if (result.resultObject[0].nextNo.isEmpty()) {

                    layout_notice_detail_next.visibility = View.GONE
                } else {

                    layout_notice_detail_next.visibility = View.VISIBLE
                    seqNextNo = result.resultObject[0].nextNo
                }

                if (result.resultObject[0].prevNo.isEmpty()) {

                    layout_notice_detail_prev.visibility = View.GONE
                } else {

                    layout_notice_detail_prev.visibility = View.VISIBLE
                    seqPrevNo = result.resultObject[0].prevNo
                }
            } else {

                layout_notice_detail.visibility = View.GONE
                layout_notice_detail_reload.visibility = View.VISIBLE
            }
        }
    }
}